### PR TITLE
Settings main screen

### LINKED ISSUE #77: Settings page
- [ ] done button
- [ ] accessed from cog in list view

## Visual Design
**InVision**
https://mozilla.invisionapp.com/share/7JEUG47ZU#/screens/267560609
**Zeplin**
https://zpl.io/VKNMd5W

### PR DESCRIPTION
**Pull request by joeyg**
resolves #77

@sashei Before I get into writing tests I thought I would ask for feedback on the architecture. Note this patch does not include the "Sign out" button which seemed like it deserved its own PR.

simulator screen shot - iphone 8 plus - 2018-03-11 at 13 54 39

### CODE REVIEW

**Comment by sashei** [comment]
hey @joeyg , this task is the next one up in my backlog. I'm planning to work on it starting tomorrow; if you can address these comments by then that'd be awesome, but otherwise I'll pick up where you left off.

**Comment by joeyg** [comment]
@Sashei I have about 3/4 of the changes made. I probably won’t have a chance to work on it until Thursday at lunch. Let me know if you would like me to push the changes so you can take over. Otherwise I’d be happy to finish the feature this week.

**Comment by sashei** [comment]
@joeyg sounds good! I can find other things to work on :)

**Comment by joeyg** [comment]
@sashei Pushed a bunch of updates to this. Very possible I missed one of your comments in all the code I moved around. Let me know if this looks more reactive.

File: lockbox-ios/Presenter/RootPresenter.swift [hunk 1]
```diff
@@ -16,6 +16,9 @@ protocol RootViewProtocol: class {
     var mainStackDisplayed: Bool { get }
     func startMainStack()
     func pushMainView(view: MainRouteAction)
+    
+    var isPresentingModal: Bool { get }
+    func dismissModal()
```
Comment (sashei) [inline lockbox-ios/Presenter/RootPresenter.swift]: Nice!!

File: lockbox-ios/Presenter/SettingsPresenter.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        self.userInfoStore = userInfoStore
+        self.routeActionHandler = routeActionHandler
+        self.userInfoActionHandler = userInfoActionHandler
+        
+        let biometricSetting = usesFaceId ? faceIdSetting : touchIdSetting
+        settings.value[1].items.insert(biometricSetting, at: settings.value[1].items.endIndex-1)
+        
+        userInfoStore.biometricLoginEnabled.subscribe(onNext: { enabled in
+            biometricSetting.isOn = enabled ?? false
+        })
```
Comment (sashei) [inline lockbox-ios/Presenter/SettingsPresenter.swift]: Still need to dispose of this subscription.

Comment (joeyg) [reply lockbox-ios/Presenter/SettingsPresenter.swift]: Whoops! I'll fix that up.

**Comment by joeyg** [comment]
@sashei I added some tests to this PR - let me know if there are some missing tests for a file or two.

**Comment by sashei** [comment]
@joeyg there are a <b>lot</b> of `swiftlint` warnings when building the `unitspecs` scheme on your branch, please fix those. If you don't have `swiftlint`, it's available via `brew`.

Unit test coverage-wise, every new & edited file should have coverage >80%. To check where you're at with text coverage, select the Report Navigator in Xcode:
screen shot 2018-03-20 at 7 40 28 am
and then look at coverage for a given test run. You can drill down from there to which lines in which files still need coverage. Large parts of the `SettingsView` and all changes to the `RootView` are untested, as well as some (IMO) testable pieces of the `SettingsPresenter`.

`RootPresenter` looks good other than the `swiftlint` warning.

**Comment by joeyg** [comment]
Thanks! I’ll take a look at those

**Comment by devinreams** [comment]
@joeyg I know @sashei already said it but I also wanted to say: thanks again for all your contributions here. We really appreciate your help.

**Comment by joeyg** [comment]
@devinreams Thanks for opportunity to contribute to the project!

**Comment by joeyg** [comment]
@sashei Updated the PR with some more test coverage and swiftlint fixes. Regarding the build failure on buildbuddy - it looks like that is building with XCode 9.0 instead of 9.2.

**Comment by devinreams** [comment]
dance

**Pull request merged**